src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java

```diff
@@ -113,18 +113,27 @@ private static Object objectDeserializer(Object value, Schema schema) {
     }
 
     private static Object unionDeserializer(Object value, Schema schema) {
-        return schema
-            .getTypes()
-            .stream()
-            .map(type -> {
-                try {
-                    return AvroDeserializer.objectDeserializer(value, type);
-                } catch (Exception e) {
-                    return  false;
-                }
-            })
-            .findFirst()
-            .orElseThrow();
+        // first, if value is null, check if the null type exist to avoid generating a NPE
+        if(value == null) {
+            if(schema.getTypes().stream().map(s -> s.getType()).anyMatch(t -> Type.NULL == (t))) {
+                return null;
+            }
+            else {
+                throw new NullPointerException("value is null but the schema type is not nullable: " + schema);
+            }
+        }
+
+        // then, evaluate each type and return the first that didn't generate an exception
+        for(var type  : schema.getTypes()) {
```

Comment:
you are retesting the null value here?

Reply:
Yes, right, it should not be an issue but ideally we should remove it from the list of types to try.

Reply:
ok don't worry, the list is really small